packages/material-ui/src/Fade/Fade.test.js

```diff
@@ -22,58 +21,104 @@ describe('<Fade />', () => {
   });
 
   it('should render a Transition', () => {
-    const wrapper = shallow(<Fade {...defaultProps} />);
-    assert.strictEqual(wrapper.name(), 'Transition');
+    const wrapper = mount(<Fade {...defaultProps} />);
+    assert.strictEqual(wrapper.find('Transition').exists(), true);
   });
 
   describe('event callbacks', () => {
-    it('should fire event callbacks', () => {
-      const events = ['onEnter', 'onEntering', 'onEntered', 'onExit', 'onExiting', 'onExited'];
```

Comment:
These tests have been useful in the past. I think that we can keep them. Sometimes, you refactor the logic and don't forward the events correctly. You can have a look at what I have done with the Collapse.test.js. In theory, it should make sure the render method doesn't shallow any events.

Reply:
What’s not being tested in the one I added to replace this that’s being tested in the shallow version?

Reply:
I was thinking of testing all the event properties `'onEnter', 'onEntering', 'onEntered', 'onExit', 'onExiting', 'onExited'`.